app/assets/javascripts/galleries/uploader.js

```diff
@@ -40,6 +42,17 @@ function bindFileInput(fileInput, form, submitButton, formData) {
     imageMaxHeight: 400,
 
     add: function(e, data) {
+      if (typeof limit !== 'undefined' && limit > 1) {
```

Comment:
Checks > 1 because of the limit is set to 1 then it's simply "don't allow multiple uploads but to allow switching them", as in icon#edit or galleries#edit. The existing maxNumberOfFiles argument does correctly turn this into a single file select so we can leave it be.